Assignment5/Q1.py

```diff
@@ -0,0 +1,150 @@
+# coding=utf-8
+"""The solution presented below takes θ(N+M) runtime where N is the number of characters in dictionary words (sum of
+them) and M is the number of dictionary words. There are two main steps performed in the given solution. First is to
+create a graph corresponding to the partial (lexicographical) order of characters in dictionary words. Second step
+creates a total order of characters (i. e. the lexicographical order or in other words ordered alphabet) which are
+stored at first in a singly linked list and then copied to the list of characters, This step uses a topological sort
+algorithm."""
+
+import SinglyLinkedList
+
+
+def get_index_of_first_differing_char(word1, word2):
+    """The method returns an index of the first character which is different in two given words.
+
+        Args:
+            word1: a string, the first word to perform comparison on
+            word2: a string, the second word to perform comparison on
+
+        Returns:
+            an integer, the index of first character which is different in two words, if the words
+            are the same or the beginning of longer word is the same as the whole shorter word, it
+            returns -1.
+    """
+    # Set length to the length of shorter word.
+    length = len(word1) if len(word1) < len(word2) else len(word2)
+    index = -1
+
+    # Loop over words, after loop is executed index will point to last character that is the same in two words.
+    for i in range(length):
+        if word1[i] == word2[i]:
```

Comment:
Now I see that it was a very clumsy way of implementing this function. It is changed now, a lot more concise 👍 thank you for pointing it out!